### ISSUE TITLE
Versions 0_18_0 x64 does not open.

### ISSUE THREAD

**Issue by PiliJ** [picotorrent/picotorrent#793]
Hi,
Versions 0_18_0 x64 (installer & portable) does not open (Win 8.1).
Error message : VCRUNTIME 140_1.dll is missing. Try to reinstall the program.
Thank you.

**Comment by bb10** [comment]
Download the latest Visual C++ 2015-2019 runtimes here: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

Download and install both x86 and x64.

**Comment by vktr** [comment]
The installer *should* install the correct VC++ runtime. Can you attach the installer logs?

**Comment by PiliJ** [comment]
Hi,
I had given up on installing it and I re-installed it today with an error... see attach !
FYI, a restore point is not created, because Rollback Rx is on the PC.
Thank you.
[PicoTorrent_20200112222105.log](https://github.com/picotorrent/picotorrent/files/4051519/PicoTorrent_20200112222105.log)

**Comment by PiliJ** [comment]
Close by error, sorry

**Comment by thec0mpler** [comment]
I'm no expert with this, but maybe the following info does help.

[This issue](https://github.com/dotnet/sdk/issues/6316) looks the same and says the reason behind the error is: "Another version of this product is already installed. Installation of this version cannot continue. To configure or remove the existing version of this product, use Add/Remove Programs on the Control Panel."

The following page also has some helpful information: https://kb.tableau.com/articles/issue/Cannot-Install-Tableau-Desktop-Error-0x80070666.

Maybe this helps?

**Comment by PiliJ** [comment]
I'm no expert too ; thank you for your suggestions thec0mpter ! 
It was the first attempt to install the soft.
As for the Visual C++ Redistributable, i' found this in programs and features ...
![Image 001](https://user-images.githubusercontent.com/33757855/73635188-dbfccc00-465a-11ea-8ebe-9d45545c5cfe.jpg)

**Comment by thec0mpler** [comment]
Nice fast response, PiliJ!

I think your Visual C++ Redistributables are fine, I have almost the same list.

In your logs you try to install 'PicoTorrent-0.16.1-x64.exe', maybe trying the new 18.0 version does help? https://github.com/picotorrent/picotorrent/releases

In your logs are the following lines:
```
Detect begin, 2 packages
Setting numeric variable 'VCREDISTVERSION' to value 1
Condition 'VCREDISTVERSION >= v14.23.27820' evaluates to false.
Detected package: VC_redist.x64.exe, state: Absent, cached: None
Detected package: PicoTorrentPackage, state: Absent, cached: None
```

Which are different from mine:
```
Detect begin, 2 packages
Setting string variable 'VCREDISTVERSION' to value 'v14.24.28127.04'
Condition 'VCREDISTVERSION >= v14.23.27820' evaluates to true.
Detected package: VC_redist.x64.exe, state: Present, cached: Complete
Detected package: PicoTorrentPackage, state: Present, cached: Complete
```

Some how you get the numeric value '1' back, instead of the string 'v14.24.28127.04'.

Could you open the register editor (press start and search for 'regedit') and go to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\14.0\VC\Runtimes\x64`? Do you have different values then the screenshot below?

![image](https://user-images.githubusercontent.com/4445369/73693546-ba412a80-46d6-11ea-8c15-5828b49bfa66.png)

**Comment by PiliJ** [comment]
Hello thec0mpler,
It's the same value ...

![Image 001](https://user-images.githubusercontent.com/33757855/73726081-a2919280-4726-11ea-9657-a754655d73fe.jpg)
I could install the new version without problem ; here is the log  
[PicoTorrent.20200204090959.log](https://github.com/picotorrent/picotorrent/files/4152097/PicoTorrent.20200204090959.log)

Curiously, however, the program appears in the start menu with the XnView icon
![Image 004](https://user-images.githubusercontent.com/33757855/73726682-e5079f00-4727-11ea-9239-98a9047b411f.jpg)
And it's impossible to change it ...
![Image 003](https://user-images.githubusercontent.com/33757855/73726774-0799b800-4728-11ea-99be-14433d8e354f.jpg)
Thank you !

**Comment by thec0mpler** [comment]
In commit 06783d8a97511076774cdd72ac69891f8f7e77d9 the code went from 'returning if the runtimes exists (true of false)' to 'returning the current installed version'. The `Result="exists"` is removed. At least, that is my guess.

The icon is a different issue. There is already an issue about that: #761. I remember I had the same issue, but somehow it went away.

As far as I know, @vktr does not want to fix old versions. Is this issue resolved for you?

**Comment by vktr** [comment]
@thec0mpler If I recall correctly the installer changed because there was an error in the version compressor, so we compared a version number against ‘true’ or ‘false’. I’ll take another look at it and see if I can spot anything. 

What I mean with “I don’t want to fix old versions” is that I don’t want to fix bugs in (for example) v0.15 and then release and maintain a 0.15.1 version with that fix. I’m very positive to fixing any issues in the current code base which will then be released in whatever version we release next (like 0.19) 😃

**Comment by PiliJ** [comment]
@thec0mpler
Issue resolved for me. Thank you again !
